pkg/spoke/controllers/statuscontroller/availablestatus_controller.go

```diff
@@ -129,7 +143,7 @@ func (c *AvailableStatusController) syncManifestWork(ctx context.Context, origin
 	manifestWork.Status.Conditions = workStatusConditions
 
 	// no work if the status of manifestwork does not change
-	if !needStatusUpdate && reflect.DeepEqual(originalManifestWork.Status.Conditions, manifestWork.Status.Conditions) {
+	if !needStatusUpdate && equality.Semantic.DeepEqual(originalManifestWork.Status.Conditions, manifestWork.Status.Conditions) {
```

Comment:
why not do a deep equal on both parts of the status you're updating here instead of computing it above in `needStatusUpdate`?

Reply:
I update the logic here so needStatusUpdate is not necessary